packages/styled-components/package.json

```diff
@@ -62,8 +62,45 @@
     "tslib": "2.5.0"
   },
   "peerDependencies": {
-    "react": ">= 16.8.0",
-    "react-dom": ">= 16.8.0"
+    "@emotion/is-prop-valid": "^1.2.1",
```

Comment:
I don't think this change will do what you think it will do they it stands:
- Listing packages as a `peerDependency` means styled-components expect the consuming application to have downloaded those dependencies beforehand, which I believe is not the case here (I might be wrong though 😄)

To achieve what you want (better dedupe on the consuming apps side) you can simply change the dependencies listed in the `dependencies` array to use the range syntax like you used here. NPM/Yarn will be smart and reuse the dependencies versions already present on the consuming side if there's a match for that range..

Reply:
That's why they're defined in both peer and regular dependencies, with peerDependenciesMeta declaring all but "react" are optional

Reply:
But since the `dependencies` array still defines fixed versions, it'll overwrite the peerDependenciesMeta, won't it? 🤔

Reply:
Tested the explicit dependency + permissive peer dep with a bunch of different pms:

Worked (installed only the higher version): yarn, pnpm
Didn't work (installed both versions): npm, bun

The tricky thing is with `react` the range is really wide (16.8 and up basically.) I could move things out of peerDependencies into devDependencies, but basically no pm automatically installs peerDependencies out of the box so users would get hit with an error to add the dependency explicitly. The goal of this was to ensure the relevant libraries are available _at least_ at the version we specify, with peerDependencies meant to pull in a greater applicable version if defined in the client repository.

Reply:
I [opened an issue](https://github.com/npm/cli/issues/7106) against npm, I think yarn and pnpm behavior is correct.

Reply:
Hmm, interesting one.. I expected PMs to make `dependencies` take precedence over `peerDependencies` as, to me, `peerDependencies` was purely informational to let consuming apps know that you expect something to be there.. However, that doesn't seem to be the behavior the accepted RFC you linked describes..

Even if they fix npm, it'll take time for people to adopt the version with the fix.. 
IMHO, if I consume a library that has a dependency: `"A": "^1.1.0"`, I would think that library was integration-tested with that package with that version, although anything in the range might work.. 

> The goal of this was to ensure the relevant libraries are available at least at the version we specify

You just precisely described (in my understanding) the `dependencies` array when used with ranges :D

Reply:
I imported `styled-components@6.1.5-rc.0` into the styled-components-website and ran `yarn`.. it did not take into account the peerDependencies array (that specifies `postcss: ^8.4.31` - should've resolved postcss to 8.4.32)..

![image](https://github.com/styled-components/styled-components/assets/16466682/49bccd00-91f0-4111-848a-5fab812b7d07)

When I removed the `peerDependencies` and set the `dependencies` to be a range (`postcss: ^8.4.0` for testing purposes), it resolved postcss to 8.4.32
![image](https://github.com/styled-components/styled-components/assets/16466682/b8395f2c-cca6-4764-bccf-a7891974a6a6)

Reply:
It's more obvious in the repo I linked in the bug ticket since that repo has postcss saved already at a higher version